api/deployment/list.go

```diff
@@ -134,28 +131,6 @@ func ListDeployments(c *gin.Context) {
 		return
 	}
 
-	dWithBs := []*library.Deployment{}
-
-	for _, deployment := range d {
-		b, _, err := database.FromContext(c).ListBuildsForDeployment(ctx, deployment, nil, 1, 3)
```

Comment:
Can we delete the database function `ListBuildsForDeployment`? Can be a follow up PR but would be nice to remove any dead code.